extensions/java-joor-dsl/deployment/src/main/java/org/apache/camel/quarkus/dsl/java/joor/deployment/JavaJoorDslProcessor.java

```diff
@@ -0,0 +1,181 @@
+import org.apache.camel.quarkus.core.deployment.spi.CamelContextBuildItem;
+import org.apache.camel.quarkus.dsl.java.joor.runtime.JavaJoorDslRecorder;
+import org.apache.camel.spi.Resource;
+import org.apache.camel.support.ResourceHelper;
+import org.apache.camel.util.FileUtil;
+import org.apache.camel.util.IOHelper;
+import org.apache.camel.util.StringHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JavaJoorDslProcessor {
+
+    private static final Logger LOG = LoggerFactory.getLogger(JavaJoorDslProcessor.class);
+    private static final String FEATURE = "camel-java-joor-dsl";
+
+    private static final Pattern PACKAGE_PATTERN = Pattern.compile(
+            "^\\s*package\\s+([a-zA-Z][\\.\\w]*)\\s*;.*$", Pattern.MULTILINE);
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep(onlyIf = NativeBuild.class)
+    void loadClassesAOT(BuildProducer<JavaJoorGeneratedClassBuildItem> generatedClass) throws Exception {
+        LOG.debug("Loading .class resources");
+        CamelMainHelper.forEachMatchingResource(
+                resource -> {
+                    if (!resource.getLocation().endsWith(".class")) {
+                        return;
+                    }
+                    try (InputStream is = resource.getInputStream()) {
+                        generatedClass
+                                .produce(new JavaJoorGeneratedClassBuildItem(asClassName(resource), resource.getLocation(),
+                                        is.readAllBytes()));
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+    }
+
+    @BuildStep(onlyIf = NativeBuild.class)
+    void compileClassesAOT(BuildProducer<JavaJoorGeneratedClassBuildItem> generatedClass,
+            CurateOutcomeBuildItem curateOutcomeBuildItem) throws Exception {
+        Map<String, Resource> nameToResource = new HashMap<>();
+        LOG.debug("Loading .java resources");
+        CompilationUnit unit = CompilationUnit.input();
+        CamelMainHelper.forEachMatchingResource(
+                resource -> {
+                    if (!resource.getLocation().endsWith(".java")) {
+                        return;
+                    }
+                    try (InputStream is = resource.getInputStream()) {
+                        String content = IOHelper.loadText(is);
+                        String name = determineName(resource, content);
+                        unit.addClass(name, content);
+                        nameToResource.put(name, resource);
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+        if (nameToResource.isEmpty()) {
+            return;
+        }
+        // Temporary workaround until being able to provide the classpath somehow to the method MultiCompile.compileUnit
```

Comment:
Could you please paste an issue or PR URL here so that future readers know what is this blocked by?